compiler/front/commands.nim

```diff
@@ -214,7 +215,124 @@ proc expectNoArg(conf: ConfigRef; switch, arg: string, pass: TCmdLinePass, info:
+    else: return ProcessNoteResult(kind: procNoteInvalidOption, switch: orig)
+
+  if i == arg.len: discard
+  elif i < arg.len and (arg[i] in {':', '='}): inc(i)
+  else: return ProcessNoteResult(kind: procNoteInvalidOption, switch: orig)
+
+  # TODO: `ReportKinds` being used for notes/groups/etc is just wrong, it
+  #        defines far more elements than one can actually control. A purpose
+  #        built enum is required.
+
+  proc findNote(noteSet: ReportKinds, groups: seq[(string, ReportKinds)]): ReportKinds =
+    # Check groups like `--hint/warning[all]` or `--hint[Performance]` (very
+    # vague term that maps onto multiple report kinds, such as "copies to
+    # sink") first, because report groups have the same string values:
+    # (`rlexLinterReport = "Name"`, `rsemLinterReport = "Name"`)
+    for (groupName, flags) in groups:
+      if cmpIgnoreStyle(groupName, id) == 0:
+        return flags
+
+    # report enums can have the same string value, indicating that they should
+    # be grouped, this is why we iterate through the set and comparing by name
+    for rk in items(noteSet - {repNone}):
+      if cmpIgnoreStyle($rk, id) == 0:
+        result.incl rk
+
+    if result == {}:
+      result = {repNone}
+
+  # unfortunately, hintUser and warningUser clash, otherwise
+  # implementation would simplify a bit
+  notes =
+    if state in {wHint, wHintAsError}:
+      findNote(repHintKinds, repHintGroups)
+    else:
+      findNote(repWarningKinds, repWarningGroups)
+
+  if notes == {repNone}:
+    return
+      if state in {wHint, wHintAsError}:
+        ProcessNoteResult(kind: procNoteInvalidHint,
+                          invalidHintOrWarning: id)
+      else:
+        ProcessNoteResult(kind: procNoteInvalidWarning,
+                          invalidHintOrWarning: id)
+
+  var val = substr(arg, i).normalize
+  if val == "":
+    val = "on"
+
+  if val notin ["on", "off"]:
+    # xxx in future work we should also allow users to have control over
+    # `foreignPackageNotes` so that they can enable
+    # `hints|warnings|warningAsErrors` for all the code they depend on.
+    return ProcessNoteResult(kind: procNoteExpectedOnOrOff, switch: arg, argVal: val)
+  else:
+    let isOn = val == "on"
+    if isOn and id.normalize == "all":
+      return ProcessNoteResult(kind: procNoteOnlyAllOffSupported, switch: arg, argVal: val)
+
+    for n in notes:
+      if n notin conf.cmdlineNotes or pass == passCmd1:
+        if pass == passCmd1:
+          conf.incl(cnCmdline, n)
+
+        conf.incl(cnModifiedy, n)
+
+        if state in {wWarningAsError, wHintAsError}:
+          conf.flip(cnWarnAsError, n, isOn)
+        else:
+          conf.flip(cnCurrent, n, isOn)
+          conf.flip(cnMainPackage, n, isOn)
+
+        if not isOn:
+          conf.excl(cnForeign, n)
+
+  result = ProcessNoteResult(kind: procNoteSuccess)
+
+proc deadprocessSpecificNote*(arg: string, state: TSpecialWord, pass: TCmdLinePass,
```

Comment:
Does the parent PR still use `deadprocessSpecificNote`, or can it be removed?

Reply:
crap... I thought I deleted that code!

Reply:
Thanks for spotting this, I'm going to push an updated removing this. 😅